app/src/main/java/org/schabi/newpipe/util/ExtractorHelper.java

```diff
@@ -50,7 +50,14 @@ private ExtractorHelper() {
         //no instance
     }
 
+    private static void checkServiceId(int serviceId) {
```

Comment:
IMO we don't need to check that. All the functions uses `NewPipe.getService`, which throws a self-explanatory error in the case of an non-existent service: "There's no service with the id = xx".

Reply:
Yes the functions use this method but deferred. The exception may thrown at any time in any thread and the calling method is not shown. 
Have a look at https://sentry.schabi.org/sentry/newpipe/issues/382/. There is no simple way to see from which fragment the serviceId is provided.

Reply:
So it this a blocking issue?